### PR TITLE
MS SQL Fix Translation of Unordered `row_number()`

### PR DESCRIPTION
**Pull request by ejneer**
Fixes `row_number` issue of #1233

### CODE REVIEW

**Comment by ejneer** [comment]
Looks like the failing test is the result of the [recent change](https://github.com/wch/r-source/commit/7dfb5b1a17778698e7f9dff555baca27a4145f1a) on r-devel to warn about unaliased Rd files

File: tests/testthat/_snaps/backend-mssql.md [hunk 1]
```diff
@@ -370,3 +370,12 @@
       OUTPUT `INSERTED`.`a`, `INSERTED`.`b` AS `b2`
       ;
 
+# `row_number()` without order is translated correctly
+
+    Code
+      mf %>% mutate(rn = row_number())
+    Output
+      <SQL>
+      SELECT `df`.*, ROW_NUMBER() OVER (ORDER BY (SELECT 1)) AS `rn`
```
Comment (mgirlich) [inline tests/testthat/_snaps/backend-mssql.md:379]: I'm amazed this works. This just feels so wrong 😄

**Comment by mgirlich** [comment]
Thanks for the PR. I'm actually not quite sure whether we actually want to support `row_number()` without any input.

**Comment by mgirlich** [comment]
Closed by https://github.com/tidyverse/dbplyr/pull/1331.

**Pull request closed without merging**